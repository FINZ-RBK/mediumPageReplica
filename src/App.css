.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}

input {
  width: 60%;
  padding: 12px 20px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-sizing: border-box;
}

input[type="submit"] {
  width: 60%;
  background-color: rgba(3, 168, 124, 1);
  color: white;
  padding: 14px 20px;
  margin: 8px 0;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}

input[type="submit"]:hover {
  background-color: rgb(2, 201, 148);
}

.summary {
  color: rgba(0, 0, 0, 0.54);
}

.form-div {
  border-radius: 5px;
  background-color: #f2f2f2;
  padding: 20px;
}

.signin {
  color: rgb(2, 201, 148);
}

.signin:hover {
  color: rgb(2, 175, 129);
  cursor: pointer;
  text-decoration: underline;
}

a {
  color: rgb(2, 201, 148);
}

a:hover {
  color: rgb(2, 175, 129);
  cursor: pointer;
  text-decoration: underline;
}

.error {
  font-weight: bold;
  color: rgb(199, 1, 1);
  font-family: "Courier New", Courier, monospace;
}
