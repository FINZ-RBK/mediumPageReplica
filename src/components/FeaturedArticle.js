import React from "react";
import 'bootstrap/dist/css/bootstrap.css';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Image from 'react-bootstrap/Image';
import Octicon, { Star } from '@primer/octicons-react'

class FeaturedArticle extends React.Component {
    constructor(props) {
        super(props);
        this.article = props.article.article;
        this.author = props.article.articleAuthor;
    }
    render() {
        return (
            <Container >
                <Row>
                    <Col md={{ span: 4, order: 12 }} sm={{ span: 12, order: 0 }} xs={{ span: 12, order: 0 }}>
                        <Row>
                            <h3>Featured stories</h3>
                        </Row>
                        <Row>
                            <h6>Today’s best stories, picked by our editors.</h6>
                        </Row>
                    </Col>
                    <Col md={{ span: 8, order: 0 }} sm={{ span: 12, order: 12 }} xs={{ span: 12, order: 12 }}>
                        <Row>
                            <Image src={(this.article.pic) ? this.article.pic : ""} fluid></Image>
                        </Row>
                        <Row>
                            <h1 style={{ fontSize: 3 + "vw" }}>{(this.article.title) ? this.article.title : ""}</h1>
                        </Row>
                        <Row>
                            <h6>{(this.article.subTitle) ? this.article.subTitle : ""}</h6>
                        </Row>
                        <Row>
                            <Col md={{ span: 1, order: 0 }}>
                                <Image src={(this.author.pic) ? this.author.pic : ""} roundedCircle height="40px" width="40px"></Image>
                            </Col>
                            <Col md={{ span: 11, order: 11 }}>
                                <Row>
                                    <p style={{ fontSize: 14 + "px" }}> {(this.author.name) ? this.author.name : ""} in Human Parts</p>
                                </Row>
                                <Row>
                                    <p style={{ fontSize: 14 + "px" }}> Nov 14 · {(this.article.readingTime) ? this.article.readingTime : ""} min read <Octicon icon={Star} size="small" verticalAlign="middle"></Octicon></p>
                                </Row>
                            </Col>
                        </Row>
                    </Col>
                </Row>
            </Container>
        );
    }
}
export default FeaturedArticle;