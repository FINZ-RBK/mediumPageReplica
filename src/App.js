import React from "react";
import "./App.css";
import FeaturedArticle from "./components/FeaturedArticle";
import TopicList from "./components/topicList";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import axios from "axios";
import AuthorizationBox from "./components/AuthorizationBox";

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      article: props.farticle
    };
  }
  componentDidMount() {
    var that = this;
    axios.get(`/articles/getFeatured`).then(res => {
      that.setState({ article: res.data });
    });
  }
  componentDidUpdate(prev) {
    if (prev.article !== this.props.article) {
      this.fetchData();
    }
  }
  render() {
    return (
      <Container>
        {/* <AuthorizationBox></AuthorizationBox> */}
        {/* Nav Bar */}
        <Row>
          <Col><iframe
            allowfullscreen
            scrolling="no"
            seamless="seamless"
            src={
              "https://navbar-module.herokuapp.com/featured/"
            }
            width="100%"
            height="100%"
            frameBorder="0"
          ></iframe></Col>

        </Row>
        {/* Featured */}
        <Row>
          <Col>
            <Row>
              <Col><FeaturedArticle
                article={this.state.article ? this.state.article : ""}
              ></FeaturedArticle></Col>

            </Row>
            <Row style={{ marginTop: 15 + "px" }}>
              <Col md={8}>
                <TopicList></TopicList>
              </Col>
              <Col md={4}></Col>
            </Row>
          </Col>
        </Row>
        {/* Latest */}

      </Container>
    );
  }
}

export default App;
