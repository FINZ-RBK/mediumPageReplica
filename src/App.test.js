import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import App from './App';
import FeaturedArticle from "./components/FeaturedArticle";
import TopicList from "./components/topicList";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
describe('App', () => {

  test('snapshot renders', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the inner FeaturedArticle', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(FeaturedArticle).length).toEqual(1);
  });
  it('renders the inner TopicList', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(TopicList).length).toEqual(1);
  });
  // integration test
  it('passes props to the featrued Article', () => {
    const wrapper = mount(<App />);
    const fArticleWrapper = wrapper.find(FeaturedArticle);
    expect(fArticleWrapper.find('h3').text()).toEqual('Featured stories')
  });
  // it('increments the counter', () => {
  //   const wrapper = mount(<App />);
  //   wrapper
  //     .find('button')
  //     .at(0)
  //     .simulate('click');
  //   const counterWrapper = wrapper.find(Counter);
  //   expect(counterWrapper.find('p').text()).toBe('1');
  // });
});
