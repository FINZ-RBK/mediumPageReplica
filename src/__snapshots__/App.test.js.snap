// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="container"
>
  <div
    className="row"
  />
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-md-4 col-sm-12 col-12 order-md-12 order-sm-0 order-0"
          >
            <div
              className="row"
            >
              <h3>
                Featured stories
              </h3>
            </div>
            <div
              className="row"
            >
              <h6>
                Today’s best stories, picked by our editors.
              </h6>
            </div>
          </div>
          <div
            className="col-md-8 col-sm-12 col-12 order-md-0 order-sm-12 order-12"
          >
            <div
              className="row"
            >
              <img
                className="img-fluid"
                src="https://miro.medium.com/max/3000/1*bVWy9X9vq8EYjt5ysP62gg.jpeg"
              />
            </div>
            <div
              className="row"
            >
              <h1
                style={
                  Object {
                    "fontSize": "3vw",
                  }
                }
              >
                The Healthy Way to Write About Pain
              </h1>
            </div>
            <div
              className="row"
            >
              <h6>
                Ask the questions you were too busy, or scared, to consider in the moment
              </h6>
            </div>
            <div
              className="row"
            >
              <div
                className="col-md-1 order-md-0"
              >
                <img
                  className="rounded-circle"
                  height="40px"
                  src="https://miro.medium.com/fit/c/80/80/1*nllGYSR4t-ORDtSOiHJacg.jpeg"
                  width="40px"
                />
              </div>
              <div
                className="col-md-11 order-md-11"
              >
                <div
                  className="row"
                >
                  <p
                    style={
                      Object {
                        "fontSize": "14px",
                      }
                    }
                  >
                     Eileen Pollack in Human Parts
                  </p>
                </div>
                <div
                  className="row"
                >
                  <p
                    style={
                      Object {
                        "fontSize": "14px",
                      }
                    }
                  >
                     Nov 14 · 5 min read 
                    <svg
                      aria-hidden="true"
                      className="octicon"
                      height={16}
                      role="img"
                      style={
                        Object {
                          "display": "inline-block",
                          "fill": "currentColor",
                          "userSelect": "none",
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 14 16"
                      width={14}
                    >
                      <path
                        d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74L14 6z"
                        fillRule="evenodd"
                      />
                    </svg>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="sc-kvZOFW kKxGbU"
    >
      <p>
        latest
      </p>
      <div
        className="sc-kjoXOD hSMTOQ"
      >
        <div
          className="sc-cHGsZl itHmRH"
        >
          <div
            className="sc-gisBJw uBDea"
          >
            What is Lorem Ipsum?
          </div>
          <div
            className="sc-kgAjT kfwjZI"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has
          </div>
          <div
            className="sc-cJSrbW eqgxet"
          >
            <a
              className="sc-fAjcbJ gtdtrb"
            >
              user
               
            </a>
             
            <a
              className="sc-caSCKo esQmcm"
            >
              in
            </a>
             
            <a
              className="sc-caSCKo esQmcm"
            >
              category
            </a>
            <img
              className="sc-hmzhuo glUmBV"
              src="https://www.freepngimg.com/download/instagram/60239-like-icons-bookmark-button-computer-facebook-instagram.png"
            />
          </div>
          <div
            className="sc-cJSrbW eqgxet"
          >
            <div
              className="sc-bwzfXH sc-bxivhb fdEfOo"
              display="flex"
            >
              <a
                className="sc-fAjcbJ gtdtrb"
                style={
                  Object {
                    "color": "#8f8c83",
                  }
                }
              >
                 Date 
              </a>
               
               . 
              <a
                className="sc-caSCKo esQmcm"
                style={
                  Object {
                    "color": "#8f8c83",
                  }
                }
              >
                 Times 
              </a>
            </div>
          </div>
        </div>
        <div
          className="sc-TOsTZ frjKZD"
        >
          <div
            className="sc-ksYbfQ bgwelu"
          >
            <img
              alt=""
              className="sc-gzVnrw iFwuYI"
              height={140}
              size={140}
              src="https://avatars.githubusercontent.com/primer"
              width={140}
            />
          </div>
        </div>
      </div>
       
    </div>
  </div>
</div>
`;
